### ISSUE TITLE
install fails on Linux Mint 17.1 (ubuntu derivative)

### ISSUE THREAD

**Issue by mlippert** [node-inspector/node-inspector#591]
When it didn't succeed as a sudo command I tried as root, but that didn't work either. The output is listed below.

```
foo@LM17-VM1 ~ $ su
Password: 
LM17-VM1 foo # npm install -g node-inspector
npm WARN optional dep failed, continuing default-browser-id@1.0.2
 
> ws@0.4.32 install /usr/lib/node_modules/node-inspector/node_modules/ws
> (node-gyp rebuild 2> builderror.log) || (exit 0)

make: Entering directory `/usr/lib/node_modules/node-inspector/node_modules/ws/build'
  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
  SOLINK_MODULE(target) Release/obj.target/bufferutil.node
  SOLINK_MODULE(target) Release/obj.target/bufferutil.node: Finished
  COPY Release/bufferutil.node
  CXX(target) Release/obj.target/validation/src/validation.o
  SOLINK_MODULE(target) Release/obj.target/validation.node
  SOLINK_MODULE(target) Release/obj.target/validation.node: Finished
  COPY Release/validation.node
make: Leaving directory `/usr/lib/node_modules/node-inspector/node_modules/ws/build'

> v8-profiler@5.2.4 install /usr/lib/node_modules/node-inspector/node_modules/v8-profiler
> node-pre-gyp install --fallback-to-build

gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/0.10.37"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/.node-gyp"
make: Entering directory `/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/build'
make: *** No rule to make target `../.node-gyp/0.10.37/common.gypi', needed by `Makefile'.  Stop.
make: Leaving directory `/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Linux 3.13.0-37-generic
gyp ERR! command "/usr/bin/nodejs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.2.4/node-v11-linux-ia32/profiler.node" "--module_name=profiler" "--module_path=/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.2.4/node-v11-linux-ia32"
gyp ERR! cwd /usr/lib/node_modules/node-inspector/node_modules/v8-profiler
gyp ERR! node -v v0.10.37
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/nodejs /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.2.4/node-v11-linux-ia32/profiler.node --module_name=profiler --module_path=/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.2.4/node-v11-linux-ia32' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/node_modules/node-pre-gyp/lib/util/compile.js:73:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:766:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:833:5)
node-pre-gyp ERR! System Linux 3.13.0-37-generic
node-pre-gyp ERR! command "node" "/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /usr/lib/node_modules/node-inspector/node_modules/v8-profiler
node-pre-gyp ERR! node -v v0.10.37
node-pre-gyp ERR! node-pre-gyp -v v0.6.4
node-pre-gyp ERR! not ok 
Failed to execute '/usr/bin/nodejs /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.2.4/node-v11-linux-ia32/profiler.node --module_name=profiler --module_path=/usr/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.2.4/node-v11-linux-ia32' (1)

> v8-debug@0.4.2 install /usr/lib/node_modules/node-inspector/node_modules/v8-debug
> node-pre-gyp install --fallback-to-build

gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/0.10.37"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/lib/node_modules/node-inspector/node_modules/v8-debug/.node-gyp"
make: Entering directory `/usr/lib/node_modules/node-inspector/node_modules/v8-debug/build'
make: *** No rule to make target `../.node-gyp/0.10.37/common.gypi', needed by `Makefile'.  Stop.
make: Leaving directory `/usr/lib/node_modules/node-inspector/node_modules/v8-debug/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Linux 3.13.0-37-generic
gyp ERR! command "/usr/bin/nodejs" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/usr/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v11-linux-ia32/debug.node" "--module_name=debug" "--module_path=/usr/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v11-linux-ia32"
gyp ERR! cwd /usr/lib/node_modules/node-inspector/node_modules/v8-debug
gyp ERR! node -v v0.10.37
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/nodejs /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v11-linux-ia32/debug.node --module_name=debug --module_path=/usr/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v11-linux-ia32' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/node-inspector/node_modules/v8-debug/node_modules/node-pre-gyp/lib/util/compile.js:73:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:766:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:833:5)
node-pre-gyp ERR! System Linux 3.13.0-37-generic
node-pre-gyp ERR! command "node" "/usr/lib/node_modules/node-inspector/node_modules/v8-debug/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /usr/lib/node_modules/node-inspector/node_modules/v8-debug
node-pre-gyp ERR! node -v v0.10.37
node-pre-gyp ERR! node-pre-gyp -v v0.6.4
node-pre-gyp ERR! not ok 
Failed to execute '/usr/bin/nodejs /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v11-linux-ia32/debug.node --module_name=debug --module_path=/usr/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v11-linux-ia32' (1)

npm ERR! v8-profiler@5.2.4 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the v8-profiler@5.2.4 install script.
npm ERR! This is most likely a problem with the v8-profiler package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls v8-profiler
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.13.0-37-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install" "-g" "node-inspector"
npm ERR! cwd /home/foo
npm ERR! node -v v0.10.37
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0
```

**Comment by chenweihan** [comment]
I make such a mistake．．．

**Comment by ddu08** [comment]
I'm having the same issues. Tried older versions of nodejs too but no luck?

**Comment by mlippert** [comment]
I did eventually get it to work. 1st I tried installing node 0.12. That still didn't work although it had worked ok on a brand new VM I had created.

So I reverted nodejs back to 0.10.25 from the Ubuntu repositories (trusty) instead of from [NodeSource](https://nodesource.com/blog/nodejs-v012-iojs-and-the-nodesource-linux-repositories). That initially didn't work until I also installed npm (1.3.10) from the Ubuntu repositories. (This is all on a Linux Mint 17.1 KDE 32bit VirtualBox VM.)

I'm not sure what the problem was or what weird state my machine was in, but after I installed node-inspector I switched back to 0.12.0 and all continued to work fine.
